### ISSUE TITLE
Don't auto-apply stash when switching branches

### ISSUE THREAD

**Issue by dkerkow** [FredrikNoren/ungit#959]
I may get something wrong here, but it is not clear to me why ungit automatically applies the stash I created on one branch to another branch on checkout. At first, I thought I might have some customized setting triggering that behavior, but I could not find anything like that.

From my understanding of git (and what I use as workflow), the idea is that you can use ´git stash´ to save your uncommited changes on Branch A, checkout another branch B, do some work, get back to Branch A, apply the stash and find everything like it was the moment you left.

Maybe also relevant: If I don't have any stashes and change branches, ungit will throw an error, reporting that theres no stash available.

**Comment by codingtwinky** [comment]
That is odd....  Ungit has a default flag [`autoStashAndPop`](autoStashAndPop) that "stash -> operation -> stash" if there are modified files.  This seems to be working fine on my end and it is super odd to me that it will throw an error when there are no changes at all.

Could you possibly share below data?
* Browser side error message you receive when you do this operation without any changes
* server side log while doing above with `logGitCommands` set to true.
* git version

Thanks,

**Comment by dkerkow** [comment]
Ok, found the config flag myself in the meantime, and setting it to false fixes the issue for me. 

But when it is enabled and I switch between branches without any local changes and without any stashes, I get the following error (german message means "No stash found"):
![auswahl_059](https://user-images.githubusercontent.com/2207124/30292295-e2409960-9735-11e7-85bd-0fc2befaa55f.png)

Log message is:
```info: Sending to keen.io: event {"launchTimeMs":1649,"version":"1.1.29","userHash":"bc7f0b2821e82784dc2b6bdc9a7395a3"}
2017-09-11T15:16:30.637Z - warn: Responding with ERROR:
{"isGitError":true,
"errorCode":"unknown",
"command":"-c color.ui=false -c core.quotepath=false -c core.pager=cat stash pop",
"workingDirectory":"/home/daniel/projects/xxx",
"error":"Kein Stash-Eintrag gefunden.\n",
"message":"Kein Stash-Eintrag gefunden.",
"stderr":"Kein Stash-Eintrag gefunden.\n",
"stdout":""}
```

Git version is 2.7.4 (Ubuntu 16.04)

Especially the config option `core.pager=cat stash pop` in the git call seems a little odd to me, as I would expect a pager and not another git call. I tried to reinstall ungit to see if it's an artifact from an early version, because I've been running ungit for quite a while now, but that didn't change anything. The issue itself is pretty old, too, but I never got to take a look after it.

**Comment by codingtwinky** [comment]
Oh wow, can you please try changing git language to english and try it?  This [article](https://askubuntu.com/questions/320661/change-gits-language-to-english-without-changing-the-locale) may help you.

Ungit does parse results of git command to detect it's states and I think your git language being german confused ungit.

**Comment by dkerkow** [comment]
You are right with your suspicion: changing language of git to english using `alias git='LANGUAGE=en_US.UTF-8 git'` does not trigger the bug. Stash is not being touched upon changing branches.

**Comment by ylecuyer** [comment]
I had the same issue. Adding `alias ungit='LANGUAGE=en_US.UTF-8 ungit'` fixes the problem.

@jung-kim maybe there should be a better handling of no stash or add git to be in english as a requirement in the README

**Comment by jung-kim** [comment]
Yes we need to add english language dependency to README as we parse git result to detect it's states.

**Comment by PhoneixS** [comment]
I don't know if it's possible but can it check the current language, change the language to English, run the command, and then restore the language?
